src/v1.ts

```diff
@@ -0,0 +1,341 @@
+    const scheduledWrapped: WrappedScheduledFunction = (
+      options: ContextOptions
+    ) => {
+      // Although in Typescript we require `options` some of our JS samples do not pass it.
+      options = options || {};
+
+      _checkOptionValidity(['eventId', 'timestamp'], options);
+      const defaultContext = _makeDefaultContext(cloudFunction, options);
+      const context = merge({}, defaultContext, options);
+
+      // @ts-ignore
+      return cloudFunction.run(context);
+    };
+    return scheduledWrapped;
+  }
+
+  if (
+    has(cloudFunction, '__trigger.httpsTrigger') &&
+    get(cloudFunction, '__trigger.labels.deployment-callable') !== 'true'
+  ) {
+    throw new Error(
+      'Wrap function is only available for `onCall` HTTP functions, not `onRequest`.'
+    );
+  }
+
+  if (!has(cloudFunction, 'run')) {
+    throw new Error(
+      'This library can only be used with functions written with firebase-functions v1.0.0 and above'
+    );
+  }
+
+  const isCallableFunction =
+    get(cloudFunction, '__trigger.labels.deployment-callable') === 'true';
+
+  let wrapped: WrappedFunction = (data: T, options: ContextOptions) => {
+    // Although in Typescript we require `options` some of our JS samples do not pass it.
+    options = options || {};
+    let context;
+
+    if (isCallableFunction) {
+      _checkOptionValidity(
+        ['app', 'auth', 'instanceIdToken', 'rawRequest'],
+        options
+      );
+      let callableContextOptions = options as CallableContextOptions;
+      context = {
+        ...callableContextOptions,
+      };
+    } else {
+      _checkOptionValidity(
+        ['eventId', 'timestamp', 'params', 'auth', 'authType', 'resource'],
+        options
+      );
+      const defaultContext = _makeDefaultContext(cloudFunction, options, data);
+
+      if (
+        has(defaultContext, 'eventType') &&
+        defaultContext.eventType !== undefined &&
+        defaultContext.eventType.match(/firebase.database/)
+      ) {
+        defaultContext.authType = 'UNAUTHENTICATED';
+        defaultContext.auth = null;
+      }
+      context = merge({}, defaultContext, options);
+    }
+
+    return cloudFunction.run(data, context);
+  };
+
+  return wrapped;
+}
+
+/** @internal */
+export function _makeResourceName(
+  triggerResource: string,
+  params = {}
+): string {
+  const wildcardRegex = new RegExp('{[^/{}]*}', 'g');
+  let resourceName = triggerResource.replace(wildcardRegex, (wildcard) => {
+    let wildcardNoBraces = wildcard.slice(1, -1); // .slice removes '{' and '}' from wildcard
+    let sub = get(params, wildcardNoBraces);
+    return sub || wildcardNoBraces + random(1, 9);
+  });
+  return resourceName;
+}
+
+function _makeEventId(): string {
+  return (
+    Math.random()
+      .toString(36)
+      .substring(2, 15) +
+    Math.random()
+      .toString(36)
+      .substring(2, 15)
+  );
+}
+
+function _checkOptionValidity(
```

Comment:
I don't mind if, in the future, we try to solve these sorts of concerns with the type system rather than with helper functions (though having both is great too)

Reply:
I agree, in the future type system checking would be better.